### PR TITLE
Update amazoncorretto for 2024 q2 release

### PR DESCRIPTION
**Pull request by rgithubli**
Update amazoncorretto for 2024 q2 release. Versions: 8.412.08.1, 11.0.23.9.1, 17.0.11.9.1, 21.0.3.9.1, 22.0.1.8.1.

### CODE REVIEW

**Comment by rgithubli** [comment]
One of the workflow failed due to `/bin/sh: can't create /etc/apk/repositories: Interrupted system call`: https://github.com/docker-library/official-images/actions/runs/8712897595/job/23900205186.

It looks like a retriable failure. Doesn't seem to have permission to retry (no button shows up). Can repo maintainer help retry then review? Thanks.

**Review by tianon — APPROVED** [review]

**Comment by rgithubli** [comment]
@tianon Thanks for the review! Would you be able to merge this pr in?

**Comment by tianon** [comment]
Yep, was just waiting for the CI kick to finish.

**Pull request merged**
